### PR TITLE
standardized layer code for more LEL layers

### LINKED ISSUE #198: Standardize Layer code in the LEL repository
# Description
LEL is a growing library with a set of layers. Currently the process of adding new layers is not easily described and repo contains redundant code for different layers.

# What can be done
- find similarities in code btw layers
- remove redundant code and pass layer type as parameter to a standard layer display code
(common function which could be used to reduce redundant code in layers)
- refactoring based on common functions

### PR DESCRIPTION
**Pull request by ananyaarun**
Fixes #198 

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below

Thanks!

### CODE REVIEW

**Comment by ananyaarun** [comment]
I am trying to optimize this a bit more and include the mapknitter layer as well
Thanks !

**Comment by ananyaarun** [comment]
Hey @sagarpreet-chadha , here is a working gif
I have created one standard file for4 layers here.
standardization

**Comment by ananyaarun** [comment]
Hey, I figured out on how to do this, Will finish this by tomorrow :)

**Comment by ananyaarun** [comment]
Hey @sagarpreet-chadha , kindly review this. I have created one standard file for 4 layers (fractracker, skytruth, odorreport and mapknitter)
here is a working gif

standardization

File: src/layercode.js [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+
+           if(this.layer == "skytruth"){
+                var redDotIcon =new L.icon.skyTruthIcon();
+                var lat = data.lat ;
+                var lng = data.lng;
+                var title = data.title ;
+                var url = data.link ;
+                var skymarker ;
+                if (!isNaN(lat) && !isNaN(lng) ){
+                  skymarker = L.marker([lat , lng] , {icon: redDotIcon}).bindPopup("<a href="+url+">" +title + "</a><br>" + "<br><strong> lat: " + lat + "</strong><br><strong> lon: " + lng + "</strong> <br><br>Data provided by <a href='http://alerts.skytruth.org/'>alerts.skytruth.org/</a>") ;
```
Comment (sagarpreet-chadha) [inline src/layercode.js]: Hi...can we structure this as we have done for mapknitter in line 136 for easy code readability .

File: src/layercode.js [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+
+           if(this.layer == "odorreport"){
+                var redDotIcon =new L.icon.odorReportIcon() ;
+                var lat = data.values["bcc29002-c4d3-4c2c-92c7-1c9032c3b0fd"][0].lat ;
+                var lng = data.values["bcc29002-c4d3-4c2c-92c7-1c9032c3b0fd"][0].lon ;
+                var title = data.title ;
+                var url = data.url ;
+                var odormarker ;
+                if (!isNaN(lat) && !isNaN(lng) ){
+                  odormarker = L.marker([lat , lng] , {icon: redDotIcon}).bindPopup(title + "<br><a href="+url+">" + url +"</a>" + "<br><strong> lat: " + lat + "</strong><br><strong> lon: " + lng + "</strong><br><br>Data provided by <a href='https://odorlog.ushahidi.io'>https://odorlog.ushahidi.io</a>") ;
```
Comment (sagarpreet-chadha) [inline src/layercode.js]: Here too 😄 !
```suggestion
            odormarker = L.marker([lat , lng] , {icon: redDotIcon}).bindPopup(title + "<br><a href="+url+">" + url+"</a>"+ "
                  <br><strong> lat: " + lat + "</strong>
                  <br><strong> lon: " + lng + "</strong>
                  <br>
                  <br>Data provided by <a href='https://odorlog.ushahidi.io'>https://odorlog.ushahidi.io</a>") ;
```

**Comment by ananyaarun** [comment]
Hey @sagarpreet-chadha I have made the code more readable.
Let me know if there is more to change !!

**Comment by sagarpreet-chadha** [comment]
Awesome :)

**Pull request merged**